Sources/UIKitNavigation/Navigation/NavigationStackController.swift

```diff
@@ -338,14 +338,19 @@
         )
         return
       }
-      stackController.path.append(.lazy(.element(value)))
+      withUITransaction(\.stackController, stackController) {
+        stackController.path.append(.lazy(.element(value)))
+      }
```

Comment:
This fixes things just for the case study, but because deep linking can happen directly from state we would need to explore a more universal solution to ship this.